src/app/user/guards/admin.guard.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot, UrlTree } from '@angular/router';
+import { Observable } from 'rxjs';
+import { UserStoreService } from '../user-store.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AdminGuard implements CanActivate {
+  constructor(private user: UserStoreService, private router: Router) { }
+  canActivate(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+      if(this.user.isAdmin$) {
+        this.router.createUrlTree(['/courses']);
+      }
+    return true;
```

Comment:
```suggestion
      if(isAdmin) {
        return true;
      }
      return this.router.createUrlTree(['/courses']);
```
I guess if user is admin we return true to allow the route. If user is not admin we return UrlTree for redirecting.

Reply:
done , thanks